components/Variation/basic_usage.rst

```diff
@@ -0,0 +1,259 @@
+         */
+        public function __construct($name)
+        {
+            $this->name = $name;
+            $this->variants = new ArrayCollection();
+            $this->options = new ArrayCollection();
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function getMasterVariant()
+        {
+            return $this->master;
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function setMasterVariant(VariantInterface $variant)
+        {
+            $this->master = $variant;
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function hasVariants()
+        {
+            if ($this->variants->isEmpty()) {
+                return false;
+            }
+
+            return true;
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function getVariants()
+        {
+            return $this->variants;
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function setVariants(Collection $variants)
+        {
+            $this->variants = $variants;
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function addVariant(VariantInterface $variant)
+        {
+            $this->variants->add($variant);
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function removeVariant(VariantInterface $variant)
+        {
+            $this->variants->removeElement($variant);
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function hasVariant(VariantInterface $variant)
+        {
+            if ($this->variants->contains($variant)) {
+                return true;
+            }
+
+            return false;
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function hasOptions()
+        {
+            if ($this->options->isEmpty()) {
+                return false;
+            }
+
+            return true;
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function getOptions()
+        {
+            return $this->options;
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function setOptions(Collection $options)
+        {
+            $this->options = $options;
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function addOption(OptionInterface $option)
+        {
+            $this->options->add($option);
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function removeOption(OptionInterface $option)
+        {
+            $this->options->removeElement($option);
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function hasOption(OptionInterface $option)
+        {
+            if ($this->options->contains($option)) {
+                return true;
+            }
+
+            return false;
+        }
+    }
```

Comment:
Maybe we could consider skipping some of the methods, if their implementation does not include anything important to show ... OK, I think we should keep them and refactor. :>